extensions/packageManager/src/client/feedModal.tsx

```diff
@@ -0,0 +1,175 @@
+          />
+      </Fragment> },
+    },
+    { key: 'column3', minWidth: 80, isResizable: false, name: '', onRender: (item: PackageSourceFeed) => {
+      if (selectedItem && item.key === selectedItem.key && !editRow)
+      return <Fragment>
+        <IconButton iconProps={{ iconName: 'Edit' }} onClick={()=>setEditRow(true)} disabled={!selectedItem || selectedItem.readonly} />
+        <IconButton iconProps={{ iconName: 'Delete' }} onClick={removeSelected} disabled={!selectedItem || selectedItem.readonly} />
+      </Fragment>;
+
+      if (selectedItem && item.key === selectedItem.key && editRow)
+      return <Fragment>
+        <IconButton iconProps={{iconName: 'Checkmark'}} onClick={() => { setEditRow(false); props.onUpdateFeed(selectedItem.key, selectedItem) }}
+                    disabled={!selectedItem || !selectedItem.text || !selectedItem.url || selectedItem.readonly}
+         />
+      </Fragment>;
+    },}
+  ];
+
+  const updateSelected = (field: string) => {
+    return (evt, val) => {
+      const newSelection = {
+        ...selectedItem,
+        [field]: val,
+      };
+      setSelectedItem(newSelection);
+    };
+  };
+
+  const addItem = () => {
+    const newItem = {
+      key: uuid(),
+      text: 'New Feed',
+      url: 'http://',
+    } as PackageSourceFeed;
+
+    const newItems = items.concat([newItem]);
+
+    // select newest to display in the editor
+    setSelectedItem(newItem);
+
+    // update the detail view
+    setItems(newItems);
+
+    // update the selection
+    selection.setItems(newItems, true);
+    selection.setKeySelected(newItem.key, true, false);
+    setSelection(selection);
+  };
+
+  const removeSelected = async () => {
+    if (await confirm(formatMessage('Delete this feed?'), formatMessage('Are you sure you want to remove this feed source?'))) {
+      setSelectedItem(undefined);
+      props.onUpdateFeed(selectedItem.key, null);
+    }
+  };
+
+  return (
+    <DialogWrapper
+      isOpen={!props.hidden}
+      dialogType={DialogTypes.Customer}
+      minWidth={900}
+      title={uitext.title}
+      subText={uitext.subTitle}
+      onDismiss={props.closeDialog}
+    >
+      <div style={{ minHeight: '300px' }} data-is-scrollable="true">
+        <DetailsList
+          items={items}
+          columns={columns}
+          setKey="set"
+          selectionMode={SelectionMode.single}
+          layoutMode={DetailsListLayoutMode.justified}
+          checkboxVisibility={CheckboxVisibility.hidden}
+          selection={selection}
+          selectionPreservedOnEmptyClick={true}
+          ariaLabelForSelectionColumn="Toggle selection"
+          ariaLabelForSelectAllCheckbox="Toggle selection for all items"
+          checkButtonAriaLabel="Row checkbox"
+        />
+        <ActionButton iconProps={{iconName: 'Add'}} onClick={addItem}>Add</ActionButton>
+        <DialogFooter>
+          <PrimaryButton onClick={props.closeDialog}>Done</PrimaryButton>
```

Comment:
localize

Reply:
is there going to be an explicit dismiss button besides the X (closing the modal)

Reply:
yes